src/itwinai/loggers.py

```diff
@@ -349,11 +349,11 @@ class ConsoleLogger(Logger):
 
     def __init__(
         self,
-        savedir: str = "mllogs",
+        savedir: Path | str = "mllogs",
         log_freq: Union[int, Literal["epoch", "batch"]] = "epoch",
         log_on_workers: Union[int, List[int]] = 0,
     ) -> None:
-        savedir = savedir = Path(savedir) / "simple-logger"
+        savedir = Path(savedir) / "simple-logger"
```

Comment:
This is fine syntactically, but I am at slight unease with the idea of setting `savedir` to something else. Not very important, but could be nice to use a different variable name instead of updating it. This is also true for some of the other functions in this file.

Reply:
yes, this made me notice that the constructor's `savedir` does not coincide with the homonymous class attribute. This would need some refactor, but it's a bit out of the scope of this PR. nice comment though